ocaml-lsp-server/src/custom_requests/req_merlin_call_compatible.ml

```diff
@@ -0,0 +1,181 @@
+
+  let uri_of_yojson params =
+    match List.assoc_opt "uri" params with
+    | Some uri -> Some (Uri.t_of_yojson uri)
+    | _ ->
+      (* If the parameter is incorrectly formatted or missing, we refuse to build
+         the parameter, [uri] is mandatory. *)
+      None
+
+  let stringish_of_yojson =
+    (* The function is relatively optimistic and attempts to treat literal data as
+       strings of characters. *)
+    function
+    | `String s -> Some s
+    | `Bool b -> Some (string_of_bool b)
+    | `Float f -> Some (string_of_float f)
+    | `Int i -> Some (string_of_int i)
+    | `Intlit i -> Some i
+    | _ -> None
+
+  let args_of_yojson_list args =
+    let open Option.O in
+    let+ args =
+      List.fold_left
+        ~f:(fun acc x ->
+          let* acc in
+          let+ x = stringish_of_yojson x in
+          x :: acc)
+        ~init:(Some [])
+        args
+    in
+    List.rev args
+
+  let args_of_yojson_assoc args =
+    let open Option.O in
+    let+ args =
+      List.fold_left
+        ~f:(fun acc (key, value) ->
+          let key = "-" ^ key in
+          let* acc in
+          let+ x = stringish_of_yojson value in
+          x :: key :: acc)
+        ~init:(Some [])
+        args
+    in
+    List.rev args
+
+  let args_of_yojson params =
+    match List.assoc_opt "args" params with
+    | Some (`List args) -> args_of_yojson_list args
+    | Some (`Assoc args) -> args_of_yojson_assoc args
+    | _ ->
+      (* If args is not a list or is absent, it should fail. *)
+      None
+
+  let t_of_yojson = function
+    | `Assoc params ->
+      let result_as_sexp = as_sexp_of_yojson params in
+      let open Option.O in
+      let* command = command_of_yojson params in
+      let* args = args_of_yojson params in
+      let* uri = uri_of_yojson params in
+      Some { result_as_sexp; command; args; uri }
+    | _ -> None
+end
+
+let raise_invalid_params ?data ~message () =
+  let open Jsonrpc.Response.Error in
+  raise @@ make ?data ~code:Code.InvalidParams ~message ()
+
+let from_structured_json_exn = function
+  | None -> raise_invalid_params ~message:"Expected params but received none" ()
+  | Some params -> (
+    match Request_params.t_of_yojson params with
+    | Some params -> params
+    | None ->
+      let data =
+        `Assoc
+          [ ("expectedParams", Request_params.expected)
+          ; ("receivedParams", (params :> Json.t))
+          ]
+      in
+      raise_invalid_params ~data ~message:"Unexpected params format" ())
+
+let with_pipeline state uri specs raw_args cmd_args f =
+  match Document_store.get_opt state.State.store uri with
+  | Some doc ->
+    let merlin = Document.merlin_exn doc in
+    let open Fiber.O in
+    let* config = Document.Merlin.mconfig merlin in
+    let specs = List.map ~f:snd specs in
+    let args =
+      Mconfig.parse_arguments
+        ~wd:(Sys.getcwd ())
+        ~warning:ignore
+        specs
+        raw_args
+        config
+        cmd_args
+      |> snd
```

Comment:
I don't think we should ignore the new configuration here: command line arguments might change the configuration for a specific query. 

I pushe a test illustrating this in 7799ef878836a476257c85b8d02cb2e5f1b2eaaf

Reply:
Thanks for the spot!

Reply:
Fixed! Thanks a lot